internal/services/eventhub/eventhub_namespace_resource.go

```diff
@@ -619,6 +619,11 @@ func resourceEventHubNamespaceDelete(d *pluginsdk.ResourceData, meta interface{}
 		return err
 	}
 
+	// need to wait for namespace status to be ready before deleting.
+	if err := waitForEventHubNamespaceStatusToBeReady(ctx, meta, *id, d.Timeout(pluginsdk.TimeoutUpdate)); err != nil {
+		return fmt.Errorf("waiting for eventHub namespace %s state to be ready error: %+v", *id, err)
+	}
```

Comment:
this indicates that another resource isn't marking itself as Created/Updated correctly, which resource is causing this?

Reply:
its disasterRecoveryConfig resource. The situation is the as the same as servicebus namespace. The fix was made in pr:https://github.com/hashicorp/terraform-provider-azurerm/pull/18706

Reply:
So is this fixed in `main`?

Reply:
Got it - so we should look to revert #18706 and then fix the actual issue here - this type of fix ends up causing larger issues elsewhere unfortunately.

Reply:
Thanks @tombuildsstuff for the comment. I left my comment in the thread, hope that explains your concern. :)